### PR TITLE
Static leaks fixed in splash screen activity

### LINKED ISSUE #1958: Static leaks in SplashScreen Activity
**Actual Behaviour**

The Async task in Splashscreen Activity is not static, and that is why is losing 3.5kb memory.

**Expected Behaviour**

The AsyncTask Class declared in SplashScreenActivity should be static for no memory dump.

**Steps to reproduce it**

1. Async task class to be made static
2. Application Context to be passed to the inner classes.

**LogCat for the issue**

In org.fossasia.phimpme:1.2.0:6.
* org.fossasia.phimpme.gallery.activities.SplashScreen has leaked:
* GC ROOT static org.fossasia.phimpme.utilities.ActivitySwitchHelper.context
* leaks org.fossasia.phimpme.gallery.activities.SplashScreen instance

* Retaining: 3.7 KB.
* Reference Key: bbeeeb59-e853-411a-b202-05937fe57f1c
* Device: motorola motorola Moto G (5) cedric_amzin
* Android Version: 7.0 API: 24 LeakCanary: 1.5.1 1be44b3
* Durations: watch=5012ms, gc=158ms, heap dump=4284ms, analysis=72153ms

* Details:
* Class org.fossasia.phimpme.utilities.ActivitySwitchHelper
|   static context = org.fossasia.phimpme.gallery.activities.SplashScreen@315000864 (0x12c68820)
|   static $classOverhead = byte[308]@318067201 (0x12f55201)
* Instance of org.fossasia.phimpme.gallery.activities.SplashScreen
|   static ACTION_OPEN_ALBUM = java.lang.String@314831744 (0x12c3f380)
|   static $classOverhead = byte[2424]@314728449 (0x12c26001)
|   static ALBUMS_PREFETCHED = 23
|   static PICK_MEDIA_REQUEST = 44
|   static CONTENT = java.lang.String@1878457912 (0x6ff6fe38)
|   static ALBUMS_BACKUP = 60
|   static PHOTOS_PREFETCHED = 2
|   static PICK_MODE = java.lang.String@314740496 (0x12c28f10)
|   PICK_INTENT = false
|   READ_EXTERNAL_STORAGE_ID = 12
|   SP = org.fossasia.phimpme.gallery.util.PreferenceUtil@314934240 (0x12c583e0)
|   album = null
|   can_be_finished = true
|   logoView = android.support.v7.widget.AppCompatImageView@318054400 (0x12f52000)
|   nextIntent = android.content.Intent@315178496 (0x12c93e00)
|   parentView = android.widget.RelativeLayout@318051328 (0x12f51400)
|   SP = org.fossasia.phimpme.gallery.util.PreferenceUtil@314934240 (0x12c583e0)
|   applyThemeImgAct = true
|   coloredNavBar = true
|   obscuredStatusBar = true
|   themeHelper = org.fossasia.phimpme.gallery.util.ThemeHelper@317827552 (0x12f1a9e0)
|   SP = org.fossasia.phimpme.gallery.util.PreferenceUtil@314934240 (0x12c583e0)
|   colors = int[2]@314818960 (0x12c3c190)
|   isSWNavBarChecked = true
|   nav_acc = android.support.design.internal.BottomNavigationItemView@317288448 (0x12e97000)
|   nav_cam = android.support.design.internal.BottomNavigationItemView@317224960 (0x12e87800)
|   nav_home = android.support.design.internal.BottomNavigationItemView@317261824 (0x12e90800)
|   navigationView = android.support.design.widget.BottomNavigationView@316666880 (0x12dff400)
|   states = int[][2]@314818984 (0x12c3c1a8)
|   mDelegate = android.support.v7.app.AppCompatDelegateImplN@315207888 (0x12c9b0d0)
|   mEatKeyUpEvent = false
|   mResources = null
|   mThemeId = 2131493197
|   mCreated = true
|   mFragments = android.support.v4.app.FragmentController@314933696 (0x12c581c0)
|   mHandler = android.support.v4.app.FragmentActivity$1@314608480 (0x12c08b60)
|   mNextCandidateRequestIndex = 0
|   mOptionsMenuInvalidated = false
|   mPendingFragmentActivityResults = android.support.v4.util.SparseArrayCompat@315233928 (0x12ca1688)
|   mReallyStopped = true
|   mRequestedPermissionsFromFragment = false
|   mResumed = false
|   mRetaining = false
|   mStopped = true
|   mStartedActivityFromFragment = false
|   mStartedIntentSenderFromFragment = false
|   mExtraDataMap = android.support.v4.util.SimpleArrayMap@314819104 (0x12c3c220)
|   mActionBar = null
|   mActionModeTypeStarting = 0
|   mActivityInfo = android.content.pm.ActivityInfo@314863968 (0x12c47160)
|   mActivityTransitionState = android.app.ActivityTransitionState@315824048 (0x12d317b0)
|   mApplication = org.fossasia.phimpme.MyApplication@315064624 (0x12c78130)
|   mCalled = true
|   mChangeCanvasToTranslucent = false
|   mChangingConfigurations = false
|   mComponent = android.content.ComponentName@314933456 (0x12c580d0)
|   mConfigChangeFlags = 0
|   mCurrentConfig = android.content.res.Configuration@315680296 (0x12d0e628)
|   mDecor = null
|   mDefaultKeyMode = 0
|   mDefaultKeySsb = null
|   mDestroyed = true
|   mDoReportFullyDrawn = false
|   mEatKeyUpEvent = false
|   mEmbeddedID = null
|   mEnableDefaultActionBarUp = false
|   mEnterTransitionListener = android.app.SharedElementCallback$1@1882879128 (0x703a

**Would you like to work on the issue?**

Yes, I would like to work on the issue.

### PR DESCRIPTION
**Pull request by DhruvamSharma**
Fixed #1958  

Changes: SplashscreenActivity Changed.
Context Field and Activity Field added in the constructor.

### CODE REVIEW

File: app/src/main/java/org/fossasia/phimpme/gallery/activities/SplashScreen.java [hunk 1]
```diff
@@ -40,13 +42,13 @@
...
     public final static int ALBUMS_BACKUP = 60;
-    private boolean PICK_INTENT = false;
+    private static boolean PICK_INTENT = false;
     public final static String ACTION_OPEN_ALBUM = "vn.mbm.phimp.leafpic.OPEN_ALBUM";
 
     //private HandlingAlbums albums;
-    private Album album;
-    private boolean can_be_finished = false;
-    private Intent nextIntent = null;
+    private static Album album;
```
Comment (abishekvashok) [inline app/src/main/java/org/fossasia/phimpme/gallery/activities/SplashScreen.java]: It's not initialized and still making it static?

Comment (DhruvamSharma) [reply app/src/main/java/org/fossasia/phimpme/gallery/activities/SplashScreen.java]: I'll make the changes.

File: app/src/main/java/org/fossasia/phimpme/gallery/activities/SplashScreen.java [hunk 2]
```diff
@@ -40,13 +42,13 @@
...
+    private static boolean can_be_finished = false;
```
Comment (yashLadha) [inline app/src/main/java/org/fossasia/phimpme/gallery/activities/SplashScreen.java]: Reference them from the context in the AsyncTask. That way you do not have to convert them to static.

Comment (DhruvamSharma) [reply app/src/main/java/org/fossasia/phimpme/gallery/activities/SplashScreen.java]: Okay sir. I am making changes now.

Comment (DhruvamSharma) [reply app/src/main/java/org/fossasia/phimpme/gallery/activities/SplashScreen.java]: Done @yashLadha  sir. I have made the requested changes in #1984

**Pull request closed without merging**